explorer/client/src/components/ownedobjects/OwnedObjects.tsx

```diff
@@ -0,0 +1,248 @@
+
+import { DefaultRpcClient as rpc } from '../../utils/internetapi/SuiRpcClient';
+import { navigateWithUnknown } from '../../utils/searchUtil';
+import { findDataFromID } from '../../utils/static/searchUtil';
+import { trimStdLibPrefix, processDisplayValue } from '../../utils/stringUtils';
+import DisplayBox from '../displaybox/DisplayBox';
+
+import styles from './OwnedObjects.module.css';
+
+type resultType = {
+    id: string;
+    Type: string;
+    Version: string;
+    display?: string;
+}[];
+
+const DATATYPE_DEFAULT: resultType = [
+    {
+        id: '',
+        Type: '',
+        Version: '',
+        display: '',
+    },
+];
+
+function OwnedObjectStatic({ objects }: { objects: string[] }) {
+    const results = objects.map((objectId) => {
+        const entry = findDataFromID(objectId, undefined);
+        return {
+            id: entry?.id,
+            Type: entry?.objType,
+            Version: entry?.version,
+            display: entry?.data?.contents?.display,
+        };
+    });
+
+    return <OwnedObjectView results={results} />;
+}
+
+function OwnecObjectInternetAPI({ objects }: { objects: string[] }) {
+    const [results, setResults] = useState(DATATYPE_DEFAULT);
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    useEffect(() => {
+        Promise.all(objects.map((objID) => rpc.getObjectInfo(objID))).then(
+            (results) => {
+                setResults(
+                    results.map(({ id, objType, version, data }) => ({
+                        id: id,
+                        Type: objType,
+                        Version: version,
+                        display: processDisplayValue(data.contents?.display),
+                    }))
+                );
+                setIsLoaded(true);
+            }
+        );
+    }, [objects]);
+
+    if (isLoaded) {
+        return <OwnedObjectView results={results} />;
+    } else {
+        return <div />;
+    }
+}
+
+function OwnedObjectView({ results }: { results: resultType }) {
+    const handlePreviewClick = useCallback(
+        (id: string, navigate: Function) => (e: React.MouseEvent) =>
+            navigateWithUnknown(id, navigate),
+        []
+    );
+    const navigate = useNavigate();
+    return (
+        <div id="ownedObjects" className={styles.ownedobjects}>
+            {results.map((entryObj, index1) => (
+                <div
+                    className={styles.objectbox}
+                    key={`object-${index1}`}
+                    onClick={handlePreviewClick(entryObj.id, navigate)}
+                >
+                    {entryObj.display !== undefined ? (
+                        <div className={styles.previewimage}>
+                            <DisplayBox
+                                display={entryObj.display}
+                                tag="imageURL"
+                            />
+                        </div>
+                    ) : (
+                        <div className={styles.previewimage} />
```

Comment:
if we don't have an image to display, i think we should instead have no html element here - this allows for more compact card presentation, like this:

<img width="470" alt="image" src="https://user-images.githubusercontent.com/14057748/162036461-f48a1064-72e6-41f7-9d80-f06d42aa8d04.png">

Reply:
If we have a Preview Card with no image next to a Preview Card with an image, because the images are of different sizes, we will get extra spacing as in the attached image. Is this OK?
 
![image](https://user-images.githubusercontent.com/11377188/162043093-cc9ff2f6-0899-4b20-8170-6785980b3e91.png)